lua/nvim-tree/view.lua

```diff
@@ -66,17 +66,60 @@ M.View = {
   }
 }
 
+---Find a rogue NvimTree buffer that might have been spawned by i.e. a session.
+---@return integer|nil
+local function find_rogue_buffer()
+  for i = 1, vim.fn.bufnr("$"), 1 do
+    if vim.fn.bufname(i) == "NvimTree" then
+      return i
+    end
+  end
+  return nil
+end
+
+---Check if the tree buffer is valid. The buffer is only valid if it exists and
+---hasn't been deleted or wiped.
+---@return boolean
+local function is_buf_valid()
+  local success, valid = pcall(function ()
+    return (
+      a.nvim_buf_is_valid(M.View.bufnr)
+      and a.nvim_buf_get_var(M.View.bufnr, "nvim_tree_buffer_ready") == 1
+      )
+  end)
+
+  if not success then
+    return false
+  else
+    return valid
+  end
+end
+
 -- set user options and create tree buffer (should never be wiped)
 function M.setup()
   M.View.auto_resize = vim.g.nvim_tree_auto_resize or M.View.auto_resize
   M.View.side = vim.g.nvim_tree_side or M.View.side
   M.View.width = vim.g.nvim_tree_width or M.View.width
 
   M.View.bufnr = a.nvim_create_buf(false, false)
+
+  if not pcall(a.nvim_buf_set_name, M.View.bufnr, 'NvimTree') then
+    -- Find the pre-existing NvimTree buffer, delete its windows then wipe it.
+    local bn = find_rogue_buffer()
+    if bn then
+      local win_ids = a.nvim_eval("win_findbuf(" .. bn .. ")")
+      for _, id in ipairs(win_ids) do
+        a.nvim_win_close(id, true)
+      end
+      a.nvim_command("bw " .. bn)
```

Comment:
use `a.nvim_buf_delete`

Reply:
Nice. Again, I'm too stuck in the vim script headspace, where "delete" and "wipe" do not mean the same thing.

Reply:
no worries :) i've always hated vimscript and learned neovim development alltogether with lua and its api so it's easier for me to think that way